client/components/site-selector/index.jsx

```diff
@@ -393,6 +393,7 @@ const SiteSelector = React.createClass( {
 				<Search
 					ref="siteSearch"
 					onSearch={ this.onSearch }
+					delaySearch={ true }
```

Comment:
Let's try it. I remember adding it for sites and it feeling slower under 50 sites (which would be the vast majority of our users). It's also possible we've done improvements to the component since then.